### PR TITLE
feat: functions package

### PR DESCRIPTION
**Pull request by AMFTech512**
Provides utility functions/classes for writing truffle edge functions.

### CODE REVIEW

File: functions/utils/serve.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { ConnInfo, serve } from "https://deno.land/std@0.167.0/http/server.ts";
+import { MyceliumClient } from "./graphql-client.ts";
+import { makeResp } from "./make-resp.ts";
+
+interface EdgeFunctionHandlerParams<RuntimeDataType = unknown> {
+  myceliumClient: MyceliumClient;
+  request: Request;
+  connInfo: ConnInfo;
```
Comment (rileymiller) [inline functions/utils/serve.ts:8]: can we avoid abbreviations like this? if I didn't have context I'd be confused with what this means

File: functions/utils/serve.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+) => Response | Promise<Response>;
+
+export function serveTruffleEdgeFunction<RuntimeDataType = unknown>(
+  handler: EdgeFunctionHandler<RuntimeDataType>,
+  options?: EdgeFunctionHandlerOptions,
+) {
+  return serve(async (request, connInfo) => {
+    const accessToken = request.headers.get("x-access-token");
+    const orgId = request.headers.get("x-org-id");
+
```
Comment (rileymiller) [inline functions/utils/serve.ts:27]: do we want to add any validation that the request is coming from our api or should 3rd parties be able to call this function?

File: functions/utils/graphql-client.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export class MyceliumClient {
+  constructor(
+    public accessToken: string,
+    public orgId: string,
+    public apiUrl = MYCELIUM_API_URL,
+  ) {}
+
+  query = <T = unknown>(
```
Comment (rileymiller) [inline functions/utils/graphql-client.ts:49]: do we want to call this something different since this client could also be used to make mutations?

Comment (rileymiller) [reply functions/utils/graphql-client.ts:49]: we could also consider using a lib like: https://deno.land/x/graphql_request@v4.1.0

File: functions/utils/serve.ts (hunk 1 above)
Comment (AMFTech512) [reply functions/utils/serve.ts:8]: Nope, it's a deno type :p

File: functions/utils/serve.ts (hunk 2 above)
Comment (AMFTech512) [reply functions/utils/serve.ts:27]: I'll add a todo

Comment (AMFTech512) [reply functions/utils/serve.ts:27]: Update: https://github.com/trufflehq/truffle-packages/pull/72/commits/6256d4dbdd866e24e4605ca469ae4e238455e019

**Pull request merged**